app/src/androidTest/java/com/sdp/movemeet/map/MainMapFragmentTest.java

```diff
@@ -95,6 +124,92 @@ public void mainMapFragment_isDisplayed() throws InterruptedException {
         onView(withId(R.id.fragment_map)).check(matches((isDisplayed())));
     }
 
+    @Test
+    public void activitiesUpdatesOnAdd() {
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        ActivitiesUpdater updater = ActivitiesUpdater.getInstance();
+        BackendActivityManager bam = new BackendActivityManager(db, "activities");
+
+        MainMapFragment mapFragment = fragmentTestRule.getFragment();
+        updater.fetchListActivities();
+        updater.updateListActivities((SupportMapFragment) mapFragment.getChildFragmentManager().findFragmentById(R.id.google_map), mapFragment);
+
+        Activity act = new Activity("activity",
+                "me",
+                "title",
+                10,
+                new ArrayList<String>(),
+                0,
+                0,
+                "desc",
+                new Date(),
+                10,
+                Sport.Running,
+                "address",
+                new Date());
+
+        bam.uploadActivity(act,
+                new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        updater.updateListActivities((SupportMapFragment) mapFragment.getChildFragmentManager().findFragmentById(R.id.google_map), mapFragment);
+                        try {
+                            Thread.sleep(2000);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                        }
+                        List<Activity> activities = updater.getActivities();
+                        Activity act_in_collection = activities.get(activities.size()-1);
+                        assertEquals(act.getActivityId(), act_in_collection.getActivityId());
+                        assertEquals(act.getAddress(), act_in_collection.getAddress());
+                        assertEquals(act.getDate(), act_in_collection.getDate());
+                        assertEquals(act.getDescription(), act_in_collection.getDescription());
+                        //assertEquals(act.getLatitude(), act_in_collection.getLatitude());
```

Comment:
I think these issues might be caused by weird ordering of the map loading / location fetching when handling `MapFragments`.

Reply:
In the previous version of my code, the currentLocation was not initialized before calling the LocationFetcher, hence it could have a null value indeed. I've added a "defaultLocation" that is always assigned to the currentLocation on creation of the fragment to prevent that from happening.